controllers/volumereplication_controller.go

```diff
@@ -174,6 +185,15 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return ctrl.Result{}, nil
 	}
 
+	// if we reach here means the request is not to delete the VR object.
```

Comment:
Instead of the split in logic here, may I suggest we check if VR is deleted at line 135?